jina/executors/indexers/vector.py

```diff
@@ -272,6 +272,10 @@ def query(self, keys: 'np.ndarray', top_k: int, *args, **kwargs) -> Tuple['np.nd
             Distance (the smaller the better) is returned, not the score.
 
         """
+        if self.size == 0:
+            query_count = len(keys)
+            result_count = 0
+            return np.empty((query_count, result_count)), np.empty((query_count, result_count))
```

Comment:
this is pretty dangerous, `numpy.empty` is like `memalloc` it assigns a memory but not initializing it, resulting in a *random* array. How can one tell if this is legit or not later?

Reply:
```
>>> import numpy as np
>>> np.empty([3, 3])
array([[ 2.00000000e+000,  2.00000000e+000, -3.95252517e-323],
       [ 0.00000000e+000,  2.12199579e-314,  0.00000000e+000],
       [ 0.00000000e+000,  0.00000000e+000,  1.77229088e-310]])
>>>
```

Reply:
Well in this case, since one of the dims is `0` it won't be anything. But I understand your general point. I guess it should be `zeros`?